crypto/evp/p_lib.c

```diff
@@ -1323,29 +1317,28 @@ size_t EVP_PKEY_get1_encoded_public_key(EVP_PKEY *pkey, unsigned char **ppub)
 {
     int rv;
 
-    if (pkey->ameth == NULL) {
-        OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
-
-        if (pkey->keymgmt == NULL || pkey->keydata == NULL)
-            return 0;
+    if (pkey != NULL && evp_pkey_is_provided(pkey)) {
```

Comment:
Just a thought, feel free to ignore.

This check is repeated a lot.  Might it be worthwhile incorporating the `pkey != NULL` inside the `evp_pkey_is_provided()` macro.

Reply:
Nope.  The macro needs to answer *one* question, "is this a provider native key", not additionally "is this a valid key", that makes the answer less clear and there's a risk of corner cases.